spec.html

```diff
@@ -4490,51 +4490,59 @@ <h2>Syntax</h2>
+          <emu-alg>
+            1. Let _a_ be MV of |StrUnsignedDecimalLiteral|.
+            1. If _a_ is 0, return *-0*<sub>𝔽</sub>.
+            1. Return RoundStringMVResult(-1 &times; _a_).
+          </emu-alg>
+          <emu-grammar>StrUnsignedDecimalLiteral ::: `Infinity`</emu-grammar>
+          <emu-alg>
+            1. Return *+&infin;*<sub>𝔽</sub>.
+          </emu-alg>
+          <emu-grammar>StrUnsignedDecimalLiteral ::: DecimalDigits `.` DecimalDigits? ExponentPart?</emu-grammar>
+          <emu-alg>
+            1. Let _a_ be MV of the first |DecimalDigits|.
+            1. If the second |DecimalDigits| is present, let _b_ be MV of the second |DecimalDigits|. Otherwise, let _b_ be 0.
+            1. If |ExponentPart| is present, let _e_ be MV of |ExponentPart|. Otherwise, let _e_ be 0.
+            1. Let _n_ be the number of code points in the second |DecimalDigits|.
+            1. Return RoundStringMVResult((_a_ + (_b_ &times; 10<sup>-_n_</sup>)) &times; 10<sup>_e_</sup>).
+          </emu-alg>
+          <emu-grammar>StrUnsignedDecimalLiteral ::: `.` DecimalDigits ExponentPart?</emu-grammar>
+          <emu-alg>
+            1. Let _a_ be MV of |DecimalDigits|.
+            1. If |ExponentPart| is present, let _e_ be MV of |ExponentPart|. Otherwise, let _e_ be 0.
+            1. Let _n_ be the number of code points in |DecimalDigits|.
+            1. Return RoundStringMVResult(_a_ &times; 10<sup>_e_-_n_</sup>).
+          </emu-alg>
+          <emu-grammar>StrUnsignedDecimalLiteral ::: DecimalDigits ExponentPart?</emu-grammar>
+          <emu-alg>
+            1. Let _a_ be MV of |DecimalDigits|.
+            1. If |ExponentPart| is present, let _e_ be MV of |ExponentPart|. Otherwise, let _e_ be 0.
+            1. Return RoundStringMVResult(_a_ &times; 10<sup>_e_</sup>).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-roundstringmvresult" aoid="RoundStringMVResult">
+          <h1>RoundStringMVResult ( _n_ )</h1>
+          <p>The abstract operation RoundStringMVResult takes argument _n_ (a mathematical value). It converts _n_ to a Number in an implementation-defined manner. For the purposes of this abstract operation, a digit is significant if it is not zero or there is a non-zero digit to its left and there is a non-zero digit to its right. It performs the following steps when called:</p>
+          <emu-alg>
+            1. If the decimal representation of _n_ has fewer than 20 significant digits in its decimal representation, return 𝔽(_n_).
+            1. Let _option1_ be the mathematical value denoted by the result of replacing each significant digit in the decimal representation of _n_ after the 20th with a 0 digit.
+            1. Let _option2_ be the mathematical value denoted by the result of replacing each significant digit in the decimal representation of _n_ after the 20th with a 0 digit and then incrementing it at the 20th position (with carrying as necessary).
```

Comment:
It's odd that this operation can 'invoke' "the mathematical value denoted by <a sequence of digits>" when MV itself has to explicitly define the rules by which that value is computed. (The current spec uses MV at the corresponding points.)

Reply:
This is again the thing we were talking past each other about in #1554. See the definition of "the mathematical value of" in [5.2.5 Mathematical Operations](https://tc39.es/ecma262/#sec-mathematical-operations). While the kind of usage here is arguably still not described (and probably should be), notice how there is no mention of the MV AO in that entire section. MV is just another AO and you should avoid associating it with the spec's internal conversions between numeric types.

Reply:
> This is again the thing we were talking past each other about in #1554.

(I don't think so. At least, it wasn't what *I* was talking about in #1554.)

> See the definition of "the mathematical value of" in [5.2.5 Mathematical Operations](https://tc39.es/ecma262/#sec-mathematical-operations). While the kind of usage here is arguably still not described (and probably should be), notice how there is no mention of the MV AO in that entire section. MV is just another AO

"The mathematical value of x" is how the spec converts a Number or BigInt value to a mathematical value, but that's not relevant to these steps because neither involves a Number or BigInt value. 

Or are you drawing an analogy? I.e., in the same way that the definition of "the mathematical value of x" doesn't refer to MV, so too the definition of RoundStringMVResult shouldn't refer to MV? I don't see how that analogy is valid.

Do you think the current spec is incorrect to invoke MV at the corresponding points in 7.1.4.1.1 and 12.8.3.2?

> and you should avoid associating it with the spec's internal conversions between numeric types.

I'm not.

Reply:
I just don't understand at all what you're talking about here. Do you agree with me that the AO named MV and the prose "the mathematical value of" are not related in any way? Is the fact that the name `MV` is composed of the first two letters of "mathematical" and "value" causing you to think that it is being used as an initialism for "mathematical value" instead of to refer to the AO with that name?

Reply:
We talked about this a bit in the editor call today. @jmdyck does the most recent commit (e61d6dc) address your concerns?

Reply:
> Is the fact that the name `MV` is composed of the first two letters of "mathematical" and "value" causing you to think that it is being used as an initialism for "mathematical value" instead of to refer to the AO with that name?

Nope.

> Do you agree with me that the AO named MV and the prose "the mathematical value of" are not related in any way?

Sure:
- MV is an AO that maps literals to mathematical values.
- "the mathematical value of" is a special phrase that 'converts' Numbers and BigInts to mathematical values.
 
And neither is defined in terms of the other.

However, what I'm talking about here is the new phrase "the mathematical value *denoted by*", which maps 'representations' to mathematical values, and thus is similar to (some of) what MV currently does. I'm saying it's odd  that RoundStringMVResult can just say "the mathematical value denoted by x" when elsewhere we explicitly do the arithmetic to accomplish the same result. It's an inconsistency in what the spec assumes as its 'base level'.

> We talked about this a bit in the editor call today. @jmdyck does the most recent commit address your concerns?

(This adds a sentence to RoundStringMVResult defining the new phrase.) That's an improvement, I guess, but no, I don't think it addresses the point I raised.

- One way to address it would be, as in the current spec, to use MV (rather than the new phrase) to get the mathematical values of the two 20-digit options. (Though that's somewhat complicated by RoundStringMVResult working with decimal representations rather than literals.)

- Another way would be to keep the new phrase and use it elsewhere instead of explicit arithmetic. (Though I don't know how well that would work out.)

But if the editors understand my point and are okay with the inconsistency, well, it's not like it'd be the only one in the spec.

Reply:
I think it's reasonable to assume that readers understand how to derive a mathematical value from its decimal representation, as is done in this PR, but since MV (the "elsewhere") is working with parse nodes - which are not normally imbued with mathematical values as a matter of background knowledge, unlike decimal representations - it warrants a more explicit definition there. (Besides which, MV has a number of other cases it needs to define explicitly anyway, so it might as well be explicit in this case.)

That is to say: the operation "take the mathematical value denoted by this decimal representation" is a reasonable thing to assume readers are familiar with. The operation "take the mathematical value denoted by this parse node" is less so, even when the parse node happens to represent a string which is the decimal representation of a number, since the parse nodes are an artifact defined by this specification.

So I don't really regard this as an inconsistency. At any rate, I'm fine with it.

Reply:
Okay, thanks.

Reply:
@jmdyck Thanks, that was much clearer. I think I understand your concerns now. My opinion on the matter matches @bakkot's.